ToDo.Droid/AndroidViewService.cs

```diff
@@ -0,0 +1,66 @@
+﻿using ToDo.Core.Services;
+using ToDo.UI;
+using Xamarin.Essentials;
+using Xamarin.Forms;
+
+namespace ToDo.Droid
+{
+    public class AndroidViewService : INativeViewService
+    {
+        public int[] GetCoordinates(VisualElement visualElement)
+        {
+            // Get the position of the StackLayout
+            var stackLayout = (StackLayout)visualElement.Parent.Parent.Parent;
+
+            var view = Xamarin.Forms.Platform.Android.Platform.GetRenderer(stackLayout).View;
+
+            var coords = new int[4];
+            view.GetLocationOnScreen(coords);
+
+            var density = view.Context.Resources.DisplayMetrics.Density;
+
+            var stackLayoutY = (int)(coords[1] / density);
+
+            // Get the position of the selected grid
+            view = Xamarin.Forms.Platform.Android.Platform.GetRenderer(visualElement).View;
+
+            view.GetLocationOnScreen(coords);
+
+            for (int i = 0; i < 4; i++)
+            {
+                coords[i] = (int)(coords[i] / density);
+            }
+
+            coords[1] = coords[1] - stackLayoutY;
+
+            return coords;
+        }
+
+        public int ValidateYPosition(int y, VisualElement grid)
+        {
+            var height = (int)(DeviceDisplay.MainDisplayInfo.Height / DeviceDisplay.MainDisplayInfo.Density);
+
+            /*
```

Comment:
In actual production code, we prefer removing old/debug code instead of commenting out, it improves readability